### DISCUSSION TITLE
How to redirect to x page component using Routes API (/api) ?

### DISCUSSION THREAD

**Discussion by 3imed-jaberi** [vercel/next.js#28501] [Help]
Question: How to redirect to x page component using Routes API ?

**Comment by delbaoliveira** [comment]
Hi @3imed-jaberi! 👋🏼 

If you mean a client-side transition, you can use the `.push()` method of the [useRouter](https://nextjs.org/docs/api-reference/next/router) hook. You may find [this example](https://github.com/vercel/next.js/blob/canary/examples/using-router/components/Header.js) useful and read more about useRouter in our [docs here](https://nextjs.org/docs/api-reference/next/router#userouter). 

If you mean redirecting in `/api` routes, you can use the `res.redirect("/new-page")` [response helper](https://nextjs.org/docs/api-routes/response-helpers).

Please let me know if you have any questions! If you share more about your use case, I can help with something more specific. 😊

**Comment by 3imed-jaberi** [comment]
Hi @delbaoliveira 👋🏼,

Thank you for quick reply, I mean redirecting in ` /api` routes to my `x-page`.

My use case: I need to redirect to re-auth page when I lost the 2-factor-auth.


```ts
// ...
      // create a session
      const session = await createSession({ req, res });

      // authenticate the user.
      const [user, status] = await authenticateUser(body);

      // check condition: when i have 202 status code --> need to redirect to double-auth page.
      if (status === 202) {
        //FIXME: should redirect !!
        return res.redirect('/reset-2fa')
      }

      // ..................... stay auth
      if (body.staySigned) {
        setStaySignedCookie(req, res);
      } else {
        deleteStaySignedCookie(req, res);
      }

      session.set("user", user);
      await session.save();

      res.json(user);
// ...
```

**Reply by delbaoliveira** [reply]
Ah, I see. Try this: 


```js
  if (status === 202) {
    res.redirect("/reset-2fa")
    return
  }
```

**Reply by 3imed-jaberi** [reply]
I found the problem cause. I work on existing project coded by other people and I try to understand what they do to fix. 
In the project, exist a wrapper for api handlers which inject the method ....

Thank you for your time <3.

**Reply by delbaoliveira** [reply]
Awesome, glad you found the answer! 😊
